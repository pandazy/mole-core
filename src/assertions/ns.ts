import { ns as pNs } from '../ns';

export function ns(context: string): string {
	return pNs(`assertions:${context}`);
}
