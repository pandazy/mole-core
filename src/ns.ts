export function ns(context: string): string {
  return `Mole:${context}`;
}
