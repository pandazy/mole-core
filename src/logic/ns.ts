import { ns as pNs } from '../ns';

export function ns(context: string): string {
	return pNs(`logic:${context}`);
}
