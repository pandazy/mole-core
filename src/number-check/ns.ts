import { ns as pNs } from '../ns';

export function ns(context: string): string {
  return pNs(`numericCheck:${context}`);
}
