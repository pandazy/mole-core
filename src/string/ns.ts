import { ns as pNs } from '../ns';

export function ns(context: string): string {
  return pNs(`string:${context}`);
}
