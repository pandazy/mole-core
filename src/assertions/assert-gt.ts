import { assertValue } from './assert-value';
import { isGt } from '../number-check/is-gt';

export const assertGt = (bar: number) => (context: string) => (value: number) =>
  assertValue<number>({
    isValid: isGt(bar),
    failureCause: `must be greater than ${bar}`,
  })(context)(value);
