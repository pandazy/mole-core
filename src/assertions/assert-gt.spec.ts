import { assertGt } from './assert-gt';
import { ns } from './ns';

describe(ns('assertGt'), () => {
  it('should assert a value greater than a number', () => {
    expect.assertions(1);

    const assertGt6 = assertGt(6)('test@arg');

    expect(() => assertGt6(3)).toThrowErrorMatchingSnapshot();
    assertGt6(7);
  });
});
