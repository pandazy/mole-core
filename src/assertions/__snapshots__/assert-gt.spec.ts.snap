// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mole:assertions:assertGt should assert a value greater than a number 1`] = `
"
[context]: test@arg
[failure cause]: must be greater than 6
[actual value]: 3
[actual value type]: Number"
`;
